web-extensions/chrome/scripts/load-bookmarks.js

```diff
@@ -1,49 +1,116 @@
-function generateBookmarkList(bookmarks) {
+const uriParams = document.location.search.split('?')[1].split('&');
+const comicName = decodeURI(uriParams[0].split('=')[1]);
+const chapter = uriParams[1].split('=')[1];
+const page = uriParams[2].split('=')[1];
+console.log(`BmcSideBar: comic ${comicName}, chapter=${chapter}, page=${page}`);
+
+function BmcMangaList() {
+    this._node = document.getElementById('manga-list');
+}
+
+BmcMangaList.prototype.onEntryClick = function(comicId) {
+    console.log('clickedOnManga!');
+    window.top.postMessage({comicId}, '*');
+}
+
+BmcMangaList.prototype.generate = function() {
+    var bookmarks = [
+        {label: "naruto",           id: 0},
+        {label: "bleach",           id: 1},
+        {label: "one piece",        id: 2},
+        {label: "goblin slayer",    id: 3},
+        {label: "hunter x hunter",  id: 4},
+    ];
+
     console.log("generating bookmark list");
     var mangaList = document.getElementById("manga-list");
     for (var i = 0; i < bookmarks.length; ++i) {
         var manga = document.createElement('div');
-        manga.innerText = bookmarks[i];
-        manga.onclick = function() {
-            clickedOnManga(this.innerText);
-        };
+        manga.innerText = bookmarks[i].label;
+        manga.onclick = () => this.onEntryClick(bookmarks[i].id);
```

Comment:
I'm not sure this is working as expected. Can you confirm the `id` sent into the `onEntryClick` function is the one of the manga everytime? In JS, closures are a bit weird and I'm afraid that `i` might have the latest value.

Reply:
Unless I'm mistaken, the object `bookmarks[i].id` is resolved when the closure is instanciated. Quick to double check with some logging though